src/bms/command.rs

```diff
@@ -217,6 +206,81 @@ impl ObjId {
         self.0[0] = self.0[0].to_ascii_uppercase();
         self.0[1] = self.0[1].to_ascii_uppercase();
     }
+
+    /// Returns whether both characters are valid Base36 characters (0-9, A-Z).
+    #[must_use]
+    pub fn is_base36(self) -> bool {
+        self.0
+            .iter()
+            .all(|c| c.is_ascii_digit() || c.is_ascii_uppercase())
+    }
+
+    /// Returns whether both characters are valid Base62 characters (0-9, A-Z, a-z).
+    #[must_use]
+    pub fn is_base62(self) -> bool {
+        self.0
+            .iter()
+            .all(|c| c.is_ascii_digit() || c.is_ascii_uppercase() || c.is_ascii_lowercase())
+    }
+
+    /// Returns an iterator over all possible ObjId values, ordered by priority:
+    /// first all Base36 values (0-9, A-Z), then remaining Base62 values.
+    ///
+    /// Total: 3843 values (excluding null "00"), with first 1295 being Base36.
+    pub fn all_values() -> impl Iterator<Item = Self> {
+        const BASE36_CHARS: &[u8] = b"0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        const BASE62_CHARS: &[u8] =
+            b"0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+        // Generate all Base36 values first (1296 values: "00" to "ZZ")
+        let base36_values = (0..36usize).flat_map(move |first_idx| {
+            (0..36usize)
+                .map(move |second_idx| Self([BASE36_CHARS[first_idx], BASE36_CHARS[second_idx]]))
+        });
+
+        // Generate all Base62 values, then filter out Base36 ones and "00"
+        let remaining_values = (0..62usize).flat_map(move |first_idx| {
+            (0..62usize)
+                .map(move |second_idx| Self([BASE62_CHARS[first_idx], BASE62_CHARS[second_idx]]))
+                .filter(move |obj_id| {
+                    // Skip "00" and Base36 values (already yielded above)
+                    !obj_id.is_null() && !obj_id.is_base36() && obj_id.is_base62()
+                })
+        });
+
+        // Chain them: first Base36 (1296 values), then remaining (2548 values)
+        // Total: 1296 + 2548 = 3844 values
+        // Skip the first Base36 value ("00") to get 1295 Base36 + 2548 remaining = 3843 total
+        base36_values.skip(1).chain(remaining_values)
+    }
+}
+
+/// Enum representing the source of a message value that can be converted to [char; 2]
+#[derive(Clone, Copy, Debug)]
+pub enum MessageValue {
+    /// Value derived from an ObjId
+    ObjId(ObjId),
+    /// Value derived from a u8
+    U8(u8),
+}
```

Comment:
Value of message command is not exhausted with this. For example, BM98 supports decimal fraction number for channel `02` (section length) like this:
```text
#00102:12.375f
```
Also, there may be more formats on message commands in future. So I think we cannot use this approach which always converting into `char`s.

Reply:
If you want to add a this type (message value that can be converted to `[char; 2]`), it lacks the responsibility too much, I thought. The ways to convert into a `String` should be implemented by each ones separately.